x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java

```diff
@@ -420,46 +420,16 @@ public void testHavingOnScalar() {
                 error("SELECT int FROM test GROUP BY int HAVING 2 < ABS(int)"));
     }
 
-    public void testInWithDifferentDataTypes_SelectClause() {
-        assertEquals("1:17: expected data type [integer], value provided is of type [keyword]",
+    public void testInWithDifferentDataTypes() {
+        assertEquals("1:8: 2nd argument of [1 IN (2, '3', 4)] must be [integer], found value ['3'] type [keyword]",
```

Comment:
If there will be a query with `IN` where there are 100 values in its list, won't the error message be hard to follow?

Reply:
Could be, yes, but currently this how we address those type of errors for all functions and I opted of consistency. I think though that now it says: `found value ['3']` for example which the user can search within the IN list and find it quickly.